celery_utils/logged_task.py

```diff
@@ -36,6 +36,14 @@ def on_retry(self, exc, task_id, args, kwargs, einfo):
         """
         Capture the exception that caused the task to be retried, if any.
         """
-        super(LoggedTask, self).on_retry(exc, task_id, args, kwargs, einfo)
         if exc is not None:
             log.warning('[{}] retried due to {}'.format(task_id, einfo.traceback))
+        super(LoggedTask, self).on_retry(exc, task_id, args, kwargs, einfo)
+
+    def on_failure(self, exc, task_id, args, kwargs, einfo):
```

Comment:
My chief concern right off the bat is that by defining both on_failure and on_retry, we're going to start getting double-entries for retry-able tasks that fail.

Could we redesign this to have an abstract base class containing the `apply_async` logging, and 2 base classes with the on_failure/on_retry logging so consumer can choose which of the 2 they want?